src/site/topic19.rst

```diff
@@ -6,13 +6,58 @@ Topic #19 --- Recursion
 Iterative Definition vs. Recursive Definition
 =============================================
 
+* Iterative definitions of things are fairly natural for us
+* Recursion on the other hand may *feel* a little less natural, but it pops up a lot in our lives and in nature
+* Debugging can sometimes feel a little recursive
+    * Ever been debugging to deal with bug ``a``
+    * Then notice that to solve it, you need to fix ``b``
+    * Then, to fix ``b``, you need to resolve ``c``
+    * Once ``c`` is fixed, you can fix ``b``
+    * Since ``b`` is fixed, you can finally finish ``a``
+
 
 Groups of People
 ----------------
 
+* Like the line of people example above, it may seem like a silly question because it is so intuitive
 
-Series of Numbers
------------------
+* A group of people is:
+    * 2 people
+    * OR 2 people
+    * OR 3 people
+    * OR 4 people
+    * OR ...
+    * ...
+    * ...
+
+* Alternatively, we can use the idea of recursion to define a group of people a different way
+* When we use recursion to define something, we use the definition of itself in it's definition
+    * Define something in terms of itself
+
+* Recursive definition of a **group of people** is:
+    * 2 people
+    * OR a **group of people** plus one more person
+
+* Based on this definition, if I ask you if 2 people are a **group of people**, the answer is clearly yes (the first point)
+* But if I ask you if 4 people is a **group of people**, we need to do some digging
+    * In order to tell you if 4 people is a **group of people**, I need to know if 3 people is a **group of people**
+        * If 3 is, then I know 4 is since 4 is then be a **group of people** plus one more person
+    * To know if 3 people is a **group of people**, I must know if 2 people is a **group of people**
+    * I know, by definition, that 2 people is, in fact, a **group of people**
+    * Therefore, I know that 3 people is a **group of people**
+    * Thus, 4 people must be a **group of people**
+
+* In the above recursive example, you will notice a **base case** and a **recursive** case
+    * The base case is something with a clear definition
+    * The recursive case is one that defines itself in terms of itself
+
+
+.. warning::
+
+    Although there is no **hard** rule saying that you need a base case, and there are examples of situations where you
```

Comment:
If you want an example to justify this point, the [SI base units](https://en.wikipedia.org/wiki/SI_base_unit#2019_redefinition_of_the_SI_base_units) are recursively defined.